### PR TITLE
Update find_EFI_Table() in uefi.py

### PR DESCRIPTION
**Pull request by frinzell**
- `find_EFI_Table()`: skip some memory ranges
- `find_EFI_Table()` code clean up

Signed-off-by: Frinzell, Aaron <aaron.frinzell@intel.com>

### CODE REVIEW

File: chipsec/hal/uefi.py [hunk 1]
```diff
@@ -1,29 +1,28 @@
-#CHIPSEC: Platform Security Assessment Framework
-#Copyright (c) 2010-2021, Intel Corporation
+# CHIPSEC: Platform Security Assessment Framework
+# Copyright (c) 2010-2021, Intel Corporation
```
Comment (dscott90) [inline chipsec/hal/uefi.py:2]: 2022?

File: chipsec/hal/uefi.py [hunk 2]
```diff
@@ -513,55 +512,70 @@ def delete_EFI_variable( self, name, guid ):
...
-                        logger().log( "[uefi] found '{}' at 0x{:016X} but doesn't look like an actual table. keep searching..".format(table_sig, table_pa) )
-                        logger().log( table_header )
+                        logger().log("[uefi] Found '{}' at 0x{:016X} but doesn't look like an actual table. Keep searching...".format(table_sig, table_pa))
+                        logger().log(table_header)
                 else:
                     isFound = True
-                    if logger().HAL: logger().log( "[uefi] found EFI table at 0x{:016X} with signature '{}'..".format(table_pa, table_sig) )
-                    table_size = struct.calcsize( EFI_TABLES[table_sig]['fmt'] )
+                    if logger().HAL:
+                        logger().log("[uefi] Found EFI table at 0x{:016X} with signature '{}'..".format(table_pa, table_sig))
```
Comment (dscott90) [inline chipsec/hal/uefi.py:563]: add third period similar to line 558

**Review by dscott90 — APPROVED** [review]

File: chipsec/hal/uefi.py [hunk 3]
```diff
@@ -513,55 +512,70 @@ def delete_EFI_variable( self, name, guid ):
...
-                    hdr = self.cs.mem.read_physical_mem( table_pa, EFI_TABLE_HEADER_SIZE )
-                table_header = EFI_TABLE_HEADER( *struct.unpack_from( EFI_TABLE_HEADER_FMT, hdr ) )
+                    hdr = self.cs.mem.read_physical_mem(table_pa, EFI_TABLE_HEADER_SIZE)
+                table_header = EFI_TABLE_HEADER(*struct.unpack_from(EFI_TABLE_HEADER_FMT, hdr))
                 # do some sanity checks on the header
-                if 0 != table_header.Reserved or                 \
-                   0 == table_header.CRC32    or                 \
-                   table_header.Revision not in EFI_REVISIONS or \
-                   table_header.HeaderSize > MAX_EFI_TABLE_SIZE:
+                is_reserved         = 0 != table_header.Reserved
```
Comment (npmitche) [inline chipsec/hal/uefi.py:552]:
```suggestion
                is_reserved         = table_header.Reserved != 0
```

File: chipsec/hal/uefi.py [hunk 4]
```diff
@@ -513,55 +512,70 @@ def delete_EFI_variable( self, name, guid ):
...
+                is_bad_crc          = 0 == table_header.CRC32
```
Comment (npmitche) [inline chipsec/hal/uefi.py:553]:
```suggestion
                is_bad_crc          = table_header.CRC32 == 0
```

**Review by npmitche — APPROVED** [review]

**Pull request merged**